### PR TITLE
refactor: sort imports in React examples

### PR DESCRIPTION
**Pull request by vursen**

### CODE REVIEW

**Comment by vursen** [comment]
Lint failed because the diff exceeded the maximum number of files it can process at once (300).

File: .eslintrc.json [hunk 1]
```diff
@@ -45,6 +45,9 @@
           [
             // Side-effects group
             "^\\u0000",
+            // React group
+            "^Frontend/demo/react-example",
+            "^react",
```
Comment (vursen) [inline .eslintrc.json:50]: Added this group to minimize the diff.

File: frontend/demo/component/datepicker/react/date-picker-basic-features.tsx [hunk 2]
```diff
@@ -1,9 +1,9 @@
+import '@vaadin/icons/';
```
Comment (web-padawan) [inline frontend/demo/component/datepicker/react/date-picker-basic-features.tsx:1]:
```suggestion
import '@vaadin/icons';
```

File: frontend/demo/component/richtexteditor/react/rich-text-editor-set-get-value.tsx [hunk 3]
```diff
@@ -1,13 +1,13 @@
...
-import { useSignal } from '@vaadin/hilla-react-signals';
 import { useSignals } from '@preact/signals-react/runtime'; // hidden-source-line
-import { RichTextEditor } from '@vaadin/react-components-pro/RichTextEditor.js';
+import { useSignal } from '@vaadin/hilla-react-signals';
 import { TextArea, type TextAreaChangeEvent } from '@vaadin/react-components/TextArea.js';
 import type {
   RichTextEditorElement,
   RichTextEditorValueChangedEvent,
 } from '@vaadin/react-components-pro/RichTextEditor.js';
+import { RichTextEditor } from '@vaadin/react-components-pro/RichTextEditor.js';
```
Comment (web-padawan) [inline frontend/demo/component/richtexteditor/react/rich-text-editor-set-get-value.tsx:10]: Could combine with `import type` above into single import.

File: frontend/demo/component/side-nav/react/side-nav-basic.tsx [hunk 4]
```diff
@@ -1,11 +1,11 @@
+import '@vaadin/icons/';
```
Comment (web-padawan) [inline frontend/demo/component/side-nav/react/side-nav-basic.tsx:1]:
```suggestion
import '@vaadin/icons';
```

File: frontend/demo/component/timepicker/react/time-picker-validation.tsx [hunk 5]
```diff
@@ -1,9 +1,9 @@
 import { reactExample } from 'Frontend/demo/react-example'; // hidden-source-line
 import React from 'react';
-import { useSignal } from '@vaadin/hilla-react-signals';
 import { useSignals } from '@preact/signals-react/runtime'; // hidden-source-line
-import { TimePicker } from '@vaadin/react-components/TimePicker.js';
+import { useSignal } from '@vaadin/hilla-react-signals';
 import type { TimePickerChangeEvent } from '@vaadin/react-components/TimePicker.js';
+import { TimePicker } from '@vaadin/react-components/TimePicker.js';
```
Comment (web-padawan) [inline frontend/demo/component/timepicker/react/time-picker-validation.tsx:6]: Could combine this with `import type` above.

File: frontend/demo/component/upload/react/upload-labelling.tsx [hunk 6]
```diff
@@ -1,8 +1,8 @@
 import { reactExample } from 'Frontend/demo/react-example'; // hidden-source-line
-import { Notification } from '@vaadin/react-components/Notification.js';
 import React, { useEffect, useRef } from 'react';
-import { Upload } from '@vaadin/react-components/Upload.js';
+import { Notification } from '@vaadin/react-components/Notification.js';
 import type { UploadFileRejectEvent } from '@vaadin/react-components/Upload.js';
+import { Upload } from '@vaadin/react-components/Upload.js';
```
Comment (web-padawan) [inline frontend/demo/component/upload/react/upload-labelling.tsx:5]: Could combine this with `import type` above.

File: frontend/demo/component/avatar/react/avatar-basic.tsx [hunk 7]
```diff
@@ -1,11 +1,11 @@
 import { reactExample } from 'Frontend/demo/react-example'; // hidden-source-line
 import React, { useEffect } from 'react';
-import { useSignal } from '@vaadin/hilla-react-signals';
 import { useSignals } from '@preact/signals-react/runtime'; // hidden-source-line
+import { useSignal } from '@vaadin/hilla-react-signals';
 import { Avatar } from '@vaadin/react-components';
+import { HorizontalLayout } from '@vaadin/react-components/HorizontalLayout';
```
[the remark this answers is not in the document]
Comment (vursen) [reply frontend/demo/component/avatar/react/avatar-basic.tsx:6]: Thanks, I'll create a separate PR for that.

**Review by web-padawan — APPROVED** [review]

**Pull request merged**